vyper/parser/events.py

```diff
@@ -172,7 +172,11 @@ def check_list_type_match(provided):  # Check list types match.
             check_list_type_match(context.vars[arg.id].typ.subtype)
             mem_offset = 0
             for _ in range(0, size):
-                arg2 = LLLnode.from_list(pos + mem_offset, typ=typ, location='memory')
+                arg2 = LLLnode.from_list(
+                    pos + mem_offset,
+                    typ=typ,
+                    location=context.vars[arg.id].location
+                )
```

Comment:
I guess now this is partially redundant with the previous branch because this handles storage. Could you provide a note that this and the previous branch could potentially be refactored into a single branch?

Reply:
I just read this function, and I actually remember having tried to refactor it in this before because...
```
holder, maxlen = pack_args_by_32(
                    holder,
                    maxlen,
                    arg2,
                    typ,
                    context,
                    placeholder + mem_offset,
                    pos=pos,
                )
                mem_offset += get_size_of_type(typ) * 32
```
Feels the same on each of the if bodies. However it's deceivingly unique in each case to warrant the duplication, additionally leaving it as is actually is way more readable as you don't have to take not what alteration occurred in `arg2` in each case :)

Reply:
Added note.